### PR TITLE
Chore/updated gh actions

### PR DESCRIPTION
**Pull request by schmidmt**

### CODE REVIEW

**Review by Swandog — CHANGES_REQUESTED** [review]
If we're going to use `upload-artifact` v4 then we should change `download-artifact` as well

File: .github/workflows/python-build-test.yaml [hunk 1]
```diff
@@ -211,24 +211,34 @@ jobs:
...
     needs: [linux]
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
 
-      - uses: actions/download-artifact@v3
+      - uses: actions/download-artifact@v4
         with:
           name: wheels
           path: dist
```
Comment (Swandog) [inline .github/workflows/python-build-test.yaml:239]: If I'm reading the [merge docs](https://github.com/actions/upload-artifact/blob/main/merge/README.md) correctly, it seems like it's preferred to just use the `pattern` and `merge-multiple` flags for `download-artifact` in the new system? But I don't know for certain.

**Review by Swandog — APPROVED** [review]
One thought, but not something to hold up over. LGTM!

File: .github/workflows/python-build-test.yaml (hunk 1 above)
Comment (schmidmt) [reply .github/workflows/python-build-test.yaml:239]: I followed the current issue for this from the devs. We also want to merge the artifacts without the publish step.

Comment (schmidmt) [reply .github/workflows/python-build-test.yaml]: If you don't like the current pattern, I can change it.

Comment (Swandog) [reply .github/workflows/python-build-test.yaml]: No I'm not opposed. I already approved it, it's good by me.

**Review by Swandog — APPROVED** [review]

**Pull request merged**